### DISCUSSION TITLE
Question about the "relative_url" toggle functionality

### DISCUSSION THREAD

**Discussion by pedrobiqua** [songkg7/o2#409] [Q&A] [answered]
**Message:**  
Hi! I’m trying to use the new feature that adds a toggle button for the `relative_url` functionality.  
However, when I turn the toggle on and run the plugin, nothing happens. I’m wondering if this might be caused by a missing configuration on my end or the way I’m adding images in Markdown.  

Currently, I add images like this:  
```
![[20241211143249.png]]
```  

My questions are:  
- Is there any additional configuration I need to adjust for the `relative_url` to work?  
- Is the way I’m adding images in Markdown correct for this functionality?  

Thanks in advance for your help! 😊

**Comment by songkg7** [comment]
Oh this is a bug 😂 Thank you for finding it.

I fixed it immediately (https://github.com/songkg7/o2/commit/98a05a510c2155c8e60ca91a770c506ecdc25c23) and it takes effect from 2.2.1 (From now on..!).

I hope it works well... 🙏

Many thanks!

**Reply by pedrobiqua** [reply]
Thank you, Song, it's working now. Thank you so much for implementing this new feature. If I think of a new feature using the plugin, I'll open an issue and implement it. I've always wanted to contribute to a project on GitHub. I'm from Brazil, and I just graduated from college.

If you ever need help implementing something, I'll be happy to assist. I'm also interested in understanding how an Obsidian plugin works and how you organize the project, particularly regarding tests, bots, GitHub Actions, and so on.

**Reply by songkg7** [reply, marked as answer]
> Thank you, Song, it's working now. Thank you so much for implementing this new feature. If I think of a new feature using the plugin, I'll open an issue and implement it. I've always wanted to contribute to a project on GitHub. I'm from Brazil, and I just graduated from college.
> 
> If you ever need help implementing something, I'll be happy to assist. I'm also interested in understanding how an Obsidian plugin works and how you organize the project, particularly regarding tests, bots, GitHub Actions, and so on.

Thank you so much for your kind words! I'm thrilled to hear that the feature is working well for you. Contributions are always welcome, and it's fantastic that you're interested in getting involved in GitHub projects.

If you'd like to learn more about how Obsidian plugins work(include o2), I recommend checking out the official documentation: https://docs.obsidian.md/Home. It's a great resource to get started.

As for GitHub Actions and bots, they're defined in the `.github` directory. Most of them are experimental and tailored for personal use, so you don’t need to worry much about them. If there's ever an issue or something that needs improvement, I'll make sure to leave a comment.

Also, just a heads-up—this project’s code isn’t particularly elegant, as I started it back when I wasn’t very familiar with TypeScript. I’m still learning and improving along the way.

Thanks again for your enthusiasm, and I look forward to seeing your contributions! 😊
